Service/Product/RecordBuilder.php

```diff
@@ -0,0 +1,817 @@
+<?php
+
+namespace Algolia\AlgoliaSearch\Service\Product;
+
+use Algolia\AlgoliaSearch\Api\RecordBuilder\RecordBuilderInterface;
+use Algolia\AlgoliaSearch\Exception\DiagnosticsException;
+use Algolia\AlgoliaSearch\Exception\ProductDeletedException;
+use Algolia\AlgoliaSearch\Exception\ProductDisabledException;
+use Algolia\AlgoliaSearch\Exception\ProductNotVisibleException;
+use Algolia\AlgoliaSearch\Exception\ProductOutOfStockException;
+use Algolia\AlgoliaSearch\Exception\ProductReindexingException;
+use Algolia\AlgoliaSearch\Exceptions\AlgoliaException;
+use Algolia\AlgoliaSearch\Helper\AlgoliaHelper;
+use Algolia\AlgoliaSearch\Helper\ConfigHelper;
+use Algolia\AlgoliaSearch\Helper\Entity\CategoryHelper;
+use Algolia\AlgoliaSearch\Helper\Entity\Product\PriceManager;
+use Algolia\AlgoliaSearch\Helper\Image as ImageHelper;
+use Algolia\AlgoliaSearch\Logger\DiagnosticsLogger;
+use Algolia\AlgoliaSearch\Service\AlgoliaConnector;
+use Magento\Bundle\Model\Product\Type as BundleProductType;
+use Magento\Catalog\Api\Data\ProductInterface;
+use Magento\Catalog\Model\Product;
+use Magento\Catalog\Model\Product\Attribute\Source\Status;
+use Magento\Catalog\Model\Product\Visibility;
+use Magento\Catalog\Model\ResourceModel\Eav\Attribute as AttributeResource;
+use Magento\CatalogInventory\Api\StockRegistryInterface;
+use Magento\ConfigurableProduct\Model\Product\Type\Configurable;
+use Magento\Framework\DataObject;
+use Magento\Framework\Event\ManagerInterface;
+use Magento\Framework\Exception\LocalizedException;
+use Magento\Framework\Exception\NoSuchEntityException;
+use Magento\Store\Model\Store;
+use Magento\Store\Model\StoreManagerInterface;
+
+class RecordBuilder implements RecordBuilderInterface
+{
+    /**
+     * @var string[]
+     */
+    protected array $attributesToIndexAsArray = [
+        'sku',
+        'color',
+    ];
+
+    public function __construct(
+        protected ManagerInterface       $eventManager,
+        protected DiagnosticsLogger      $logger,
+        protected Visibility             $visibility,
+        protected StoreManagerInterface  $storeManager,
+        protected ConfigHelper           $configHelper,
+        protected CategoryHelper         $categoryHelper,
+        protected AlgoliaHelper          $algoliaHelper,
+        protected ImageHelper            $imageHelper,
+        protected StockRegistryInterface $stockRegistry,
+        protected PriceManager           $priceManager,
```

Comment:
I notice you that like using the trailing commas that are now allowed in PHP 8 - is that intentional? For consistency we should agree on best approach and maybe start putting adhering to a coding style guide ... 🤔

Reply:
Not really intentional no, do you prefer that we don't use trailing commas at all ?

Reply:
I'm fine either way now that we're no longer supporting PHP 7. I was just thinking maybe we should defined a style guide for stuff. Let's chat about it later. Not a blocker at all! 🙂